baseapp/grpcserver.go

```diff
@@ -80,7 +88,25 @@ func (app *BaseApp) RegisterGRPCServer(server gogogrpc.Server) {
 			}
 		}()
 
-		return handler(grpcCtx, req)
+		// Extract the existing span context from the incoming request
+		parentCtx := otel.GetTextMapPropagator().Extract(grpcCtx, propagation.HeaderCarrier(md))
+
+		// Start a new span representing the request
+		// The span ends when the request is complete
+		grpcCtx, span := tracer.Start(parentCtx, grpcInfo.FullMethod, trace.WithSpanKind(trace.SpanKindServer))
+		defer span.End()
+
+		span.SetAttributes(attribute.String("http.method", grpcInfo.FullMethod))
+
+		resp, err = handler(grpcCtx, req)
+
+		if err != nil {
+			span.SetStatus(otelcodes.Error, err.Error())
+		} else {
+			span.SetStatus(otelcodes.Ok, "OK")
+		}
```

Comment:
The span status is also set in the stats handler. Can you elaborate why you need this here?